code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/dynamic/DynamicSchemaFactory.java

```diff
@@ -61,12 +63,14 @@ public Schema getSchema(Class<?> clz) {
 
     @Override
     public Schema getSchema(String type) {
+        if (type == null) {
```

Comment:
It seems like this method could really benefit from caching.
Let me lay out the situation:
1. This is called any time a user's request needs a dynamic schema. Machine schemas now dynamic, so it is much more frequent than it used to be.
2. This method has a db hit and a conversion from a json string to an object, which (roughly speaking) arent cheap
3. The cache would need to invalidate entries when new dynamic schemas were uploaded.

Worth it @ibuildthecloud or over-engineering?